--- ./hadoop-2.1.0-beta-src-orig/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/main/java/org/apache/hadoop/mapreduce/v2/app/commit/CommitterEventHandler.java	2013-07-30 07:06:57.000000000 -0500
+++ ./hadoop-2.1.0-beta-src/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/main/java/org/apache/hadoop/mapreduce/v2/app/commit/CommitterEventHandler.java	2017-11-19 19:41:56.000000000 -0600
@@ -249,6 +249,11 @@
       try {
         touchz(startCommitFile);
         jobCommitStarted();
+java.io.File tmp = new java.io.File("/tmp/MR5-crash.flag");
+if (tmp.exists()) { 
+  tmp.delete();
+  Runtime.getRuntime().halt(0);
+}
         waitForValidCommitWindow();
         committer.commitJob(event.getJobContext());
         touchz(endCommitSuccessFile);
